package/base-files/files/lib/functions/ipv4.sh

```diff
@@ -0,0 +1,146 @@
+uint_max=4294967295
+
+# return a count of the number of bits set in $1
+bitcount() {
+    local c="$1"
+
+    c=$((((c >> 1) & 0x55555555) + (c & 0x55555555)))
+    c=$((((c >> 2) & 0x33333333) + (c & 0x33333333)))
+    c=$((((c >> 4) & 0x0f0f0f0f) + (c & 0x0f0f0f0f)))
+    c=$((((c >> 8) & 0x00ff00ff) + (c & 0x00ff00ff)))
+    c=$((((c >> 16) & 0x0000ffff) + (c & 0x0000ffff)))
+
+    echo "$c"
+}
+
+# check that $1 is only base 10 digits, and that it doesn't
+# exceed 2^32-1
+check_uint32() {
+    local n="$1"
+
+    if [ -n "${n//[0-9]/}" ]; then
+	echo "Not a decimal integer ($n)" >&2
+	exit 1
+    fi
+
+    if [ "$n" -gt $uint_max ]; then
+	echo "Out of range ($n)" >&2
+	exit 1
+    fi
+
+    echo "$n"
+}
+
+# tedious but portable with busybox's limited shell
+# we check each octet to be in the range of 0..255,
+# and also make sure there's no extaneous characters
```

Comment:
`# and also make sure there are no extraneous characters`

Reply:
Changed